### PR TITLE
feat(shorebird_cli): add new flavors to shorebird.yaml if detected by `shorebird init`

### LINKED ISSUE #1179: fix: shorebird init force replacing the old app id
**Description**

I have a shorebird.yaml and i want to regenerate the shorebird because there is a new flavor, but the app id for each flavor is changed , the flavor it should be only added a new one and not replacing the old one

**Steps To Reproduce**

1. type ```shorebird init```
2. after creating the shorebird.yaml then try to recreate the shorebird.yaml with --force
3. you will see the app_id and flavors app id is changed too

**Expected Behavior**

if shorebird.yaml  exists, app_id and flavors id that exists should not change

**Screenshots**

Old shorebird.yaml
```
app_id: dbdc3616-8bd1-4bb0-be85-8e8a41918950
flavors:
  alampangrango: dbdc3616-8bd1-4bb0-be85-8e8a41918950
  alifalittifaq: 467c8310-e247-4724-a499-a65564574328
  dev: 8cbded45-a405-4b80-b393-c81bbc9285d8
  maassunnah: 37d0c220-0218-44f4-97dd-d6aae45d9cf9
  mtsalinayah: bc9910c5-b80a-45f5-a5e8-c739fda173e4
  mtsassunnah: b142454a-2dd4-4361-8850-21712ee63689
  mtsdarulhuda: 992d8a82-e36d-40c9-b9fb-2f7c9cff4e9c
  saqnurmadani: 57e0a8a3-0154-48e6-8787-fb831a92f746
  sditalfalahcirebon: 94a93692-75c7-4cc7-b890-97b5424cb3da
  sditaliya: be26bd1c-3d92-4ff6-ba6f-2e6ef0babe1e
```

New shorebird.yaml
```
app_id: 1bf70f50-3239-4e80-af8d-3e6f9855c5fd
flavors:
  alampangrango: 1bf70f50-3239-4e80-af8d-3e6f9855c5fd
  alifalittifaq: 83a48cdd-354c-4fbe-98d5-55be2ff2f2af
  dev: 7c0005e6-0850-41d4-8b79-e0f4b1ef609d
  maassunnah: 1eed0c51-f43d-494c-bc08-447c7ae6b7ca
  mtsalifalittifaq: 5c962a33-8f6a-43a2-b683-a9ff935b7f68
  mtsalinayah: ed9a5c0f-d05a-4020-8b5c-bf371b655d1a
  mtsassunnah: c41fa1a9-2c4f-4474-9c91-b99c71c2ddb5
  mtsdarulhuda: cb91f848-3240-461d-a263-63a921496b9f
  saqnurmadani: 5767e222-730b-4e10-bc3f-511b1cb1b854
  sditalfalahcirebon: 99bec714-e3f2-400b-ac74-7b653e1c25c2
  sditaliya: e10dbeee-500f-4274-802e-38082522dc56
```

**Additional Context**

### PR DESCRIPTION
**Pull request by bryanoltman**
## Description

If new flavors are detected when running `shorebird init`, add those flavors to the `shorebird.yaml` file.

Fixes https://github.com/shorebirdtech/shorebird/issues/1179

## Type of Change

- [x] ✨ New feature (non-breaking change which adds functionality)
- [ ] 🛠️ Bug fix (non-breaking change which fixes an issue)
- [ ] ❌ Breaking change (fix or feature that would cause existing functionality to change)
- [ ] 🧹 Code refactor
- [ ] ✅ Build configuration change
- [ ] 📝 Documentation
- [ ] 🗑️ Chore

### CODE REVIEW

File: packages/shorebird_cli/lib/src/commands/init_command.dart [hunk 1]
```diff
@@ -92,6 +86,59 @@ If you want to reinitialize Shorebird, please run "shorebird init --force".''');
...
+        shorebirdYaml.appId,
+        flavors: flavorsToAppIds,
+      );
+      updateShorebirdYamlProgress.complete('Flavors added to shorebird.yaml');
+      return ExitCode.success.code;
+    }
+
+    if (!force && shorebirdEnv.hasShorebirdYaml) {
+      logger.err('''
+A "shorebird.yaml" already exists.
```
Comment (eseidel) [inline packages/shorebird_cli/lib/src/commands/init_command.dart]: and seems up to date?

**Review by eseidel — COMMENTED** [review]
Felix should probably weigh in.  We'll also need to update our docs.

**Review by felangel — APPROVED** [review]

File: packages/shorebird_cli/lib/src/commands/init_command.dart [hunk 2]
```diff
@@ -13,6 +13,7 @@ import 'package:shorebird_cli/src/platform.dart';
 import 'package:shorebird_cli/src/shorebird_env.dart';
 import 'package:shorebird_cli/src/shorebird_validator.dart';
 import 'package:shorebird_cli/src/xcodebuild.dart';
+import 'package:shorebird_code_push_client/shorebird_code_push_client.dart';
```
Comment (felangel) [inline packages/shorebird_cli/lib/src/commands/init_command.dart:16]: why is this import needed?

File: packages/shorebird_cli/lib/src/commands/init_command.dart [hunk 3]
```diff
@@ -92,6 +86,59 @@ If you want to reinitialize Shorebird, please run "shorebird init --force".''');
       return ExitCode.software.code;
     }
 
+    final shorebirdYaml = shorebirdEnv.getShorebirdYaml();
+    Set<String> newFlavors;
+    if (shorebirdYaml?.flavors != null) {
```
Comment (felangel) [inline packages/shorebird_cli/lib/src/commands/init_command.dart]: minor but might be nice to declare a variable:

```dart
final existingFlavors = shorebirdYaml?.flavors;
if (existingFlavors != null) {...}
```

To avoid having to force unwrap.

File: packages/shorebird_cli/lib/src/commands/init_command.dart (hunk 2 above)
Comment (bryanoltman) [reply packages/shorebird_cli/lib/src/commands/init_command.dart:16]: For the AppMetadata type on line 109

File: packages/shorebird_cli/lib/src/commands/init_command.dart [hunk 4]
```diff
@@ -92,6 +86,59 @@ If you want to reinitialize Shorebird, please run "shorebird init --force".''');
...
+      final existingFlavors = shorebirdYaml!.flavors!.keys.toSet();
+      newFlavors = productFlavors
+          .where((element) => !existingFlavors.contains(element))
+          .toSet();
```
Comment (felangel) [inline packages/shorebird_cli/lib/src/commands/init_command.dart]: Couldn't we use `Set.difference` here?

File: packages/shorebird_cli/lib/src/commands/init_command.dart [hunk 5]
```diff
@@ -92,6 +86,59 @@ If you want to reinitialize Shorebird, please run "shorebird init --force".''');
...
+        flavors: flavorsToAppIds,
+      );
+      updateShorebirdYamlProgress.complete('Flavors added to shorebird.yaml');
+      return ExitCode.success.code;
+    }
+
+    if (!force && shorebirdEnv.hasShorebirdYaml) {
+      logger.err('''
+A "shorebird.yaml" already exists.
+If you want to reinitialize Shorebird, please run "shorebird init --force".''');
```
Comment (felangel) [inline packages/shorebird_cli/lib/src/commands/init_command.dart]: minor but might be nice to split the recommendation into a `.info` and wrap the command in `lightCyan` for consistency

File: packages/shorebird_cli/lib/src/commands/init_command.dart (hunk 4 above)
Comment (bryanoltman) [reply packages/shorebird_cli/lib/src/commands/init_command.dart]: Done.

File: packages/shorebird_cli/lib/src/commands/init_command.dart (hunk 3 above)
Comment (bryanoltman) [reply packages/shorebird_cli/lib/src/commands/init_command.dart]: Done

File: packages/shorebird_cli/lib/src/commands/init_command.dart (hunk 5 above)
Comment (bryanoltman) [reply packages/shorebird_cli/lib/src/commands/init_command.dart]: Done

File: packages/shorebird_cli/lib/src/commands/init_command.dart (hunk 1 above)
Comment (bryanoltman) [reply packages/shorebird_cli/lib/src/commands/init_command.dart]: Done
